src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php

```diff
@@ -155,7 +155,8 @@ public function setMainAccount(ContactEntity $contact, $data)
         // set account relation
         if (isset($data['account']) &&
             isset($data['account']['id']) &&
-            $data['account']['id'] != 'null'
+            $data['account']['id'] != 'null'&&
+            !$this->getMainAccountContact($contact)
```

Comment:
this will cause that account cannot be changed anymore, once a main account-contact relation is set.